src/uu/ls/src/dired.rs

```diff
@@ -0,0 +1,178 @@
+// This file is part of the uutils coreutils package.
+//
+// For the full copyright and license information, please view the LICENSE
+// file that was distributed with this source code.
+// spell-checker:ignore dired subdired
+
+use crate::Config;
+use std::fmt;
+use std::io::{BufWriter, Stdout, Write};
+use uucore::error::UResult;
+
+#[derive(Debug, Clone)]
+pub struct BytePosition {
+    pub start: usize,
+    pub end: usize,
+}
+
+/// Represents the output structure for DIRED, containing positions for both DIRED and SUBDIRED.
+#[derive(Debug, Clone, Default)]
+pub struct DiredOutput {
+    pub dired_positions: Vec<BytePosition>,
+    pub subdired_positions: Vec<BytePosition>,
+    pub just_printed_total: bool,
```

Comment:
I'm struggling with this field, it somehow doesn't fit to the other fields because it doesn't contain any data like the other fields. Though I don't know where it should be instead :|

Reply:
yeah, i used to have it in BytePosition but terts didn't like it and suggested this place :)

However, I would not overthink this, this feature is probably not used much and won't get many updates